clay/http.py

```diff
@@ -20,39 +29,66 @@
 DEFAULT_CA_CERTS = '/etc/ssl/certs/ca-certificates.crt'
 
 
-class VerifiedHTTPSOpener(urllib2.HTTPSHandler):
+class VerifiedHTTPSOpener(HTTPSHandler):
     def https_open(self, req):
         ca_certs = config.get('http.ca_certs_file', DEFAULT_CA_CERTS)
         if config.get('http.verify_server_certificates', True) and os.path.exists(ca_certs):
-            frags = urlparse.urlparse(req.get_full_url())
+            frags = urlparse(req.get_full_url())
             ssl.get_server_certificate(
                 (frags.hostname, frags.port or 443),
                 ca_certs=ca_certs
             )
-        return self.do_open(httplib.HTTPSConnection, req)
-
-urllib2.install_opener(urllib2.build_opener(VerifiedHTTPSOpener))
-
-
-class Request(urllib2.Request):
-    '''
-    This subclass adds "method" to urllib2.Request
-    '''
-    def __init__(self, url, data=None, headers={}, origin_req_host=None,
-                 unverifiable=False, method=None):
-        urllib2.Request.__init__(self, url, data, headers, origin_req_host,
-                                 unverifiable)
-        if headers is None:
-            self.headers = {}
-        self.method = method
-
-    def get_method(self):
-        if self.method is not None:
-            return self.method
-        if self.has_data():
-            return 'POST'
-        else:
-            return 'GET'
+        return self.do_open(http.client.HTTPSConnection, req)
+
+
+install_opener(build_opener(VerifiedHTTPSOpener))
+
+if six.PY2:
+    class Request(urllib_Request):
+        '''
+        This subclass adds "method" to urllib2.Request
+        '''
+        def __init__(self, url, data=None, headers={}, origin_req_host=None,
+                     unverifiable=False, method=None):
+            urllib_Request.__init__(self, url, data, headers, origin_req_host,
+                                     unverifiable)
+            if headers is None:
+                self.headers = {}
+            self.method = method
+
+        def get_method(self):
+            if self.method is not None:
+                return self.method
+            if self.data is not None:
+                return 'POST'
+            else:
+                return 'GET'
+
+
+if six.PY3:
```

Comment:
This segment feels redundant but I can't think of a better way to improve the class definition